scripts/irods/test/test_quotas.py

```diff
@@ -228,3 +228,68 @@ def acRescQuotaPolicy(rule_args, callback, rei):
             self.admin.run_icommand(['iadmin', 'rmresc', resc_name])
             self.admin.run_icommand(['iadmin', 'rmgroup', f'{group_name}_1'])
             IrodsController().reload_configuration()
+
+    # The following test covers case 3, 4, 5
+    def test_physical_resource_with_coordinating_resources_issue_8667(self):
+        pep_map = {
+            'irods_rule_engine_plugin-irods_rule_language': textwrap.dedent('''
+                acRescQuotaPolicy {
+                    msiSetRescQuotaPolicy("on");
+                }
+            '''),
+            'irods_rule_engine_plugin-python': textwrap.dedent('''
+                def acRescQuotaPolicy(rule_args, callback, rei):
+                    callback.msiSetRescQuotaPolicy('on')
+            ''')
+        }
+
+        try:
+            with temporary_core_file() as core:
+                core.add_rule(pep_map[self.plugin_name])
+                IrodsController().reload_configuration()
+
+                group_name = 'test_group_case_3_4_5'
+                file_name = 'test_file_case_3_4_5'
+                resc_name = 'ufs0_quota_case_3_4_5'
+                other_resc_name = 'ufs1_quota_case_3_4_5'
+                repl_resc_name = 'repl_quota_case_3_4_5'
+                lib.make_file(os.path.join(self.quota_user.local_session_dir, file_name), 4096, contents='arbitrary')
+                lib.create_ufs_resource(self.admin, resc_name)
+                lib.create_ufs_resource(self.admin, other_resc_name)
+                lib.create_replication_resource(self.admin, repl_resc_name)
+
```

Comment:
please add a passthru resource 'above' the replication resource... this will confirm parent-parent-resc relationships are still honored and 'rolled up' correctly.